static/app/views/dashboardsV2/detail.tsx

```diff
@@ -87,18 +90,43 @@ class DashboardDetail extends Component<Props, State> {
     this.checkStateRoute();
     router.setRouteLeaveHook(route, this.onRouteLeave);
     window.addEventListener('beforeunload', this.onUnload);
+    this.checkIfShouldMountWidgetViewerModal();
   }
 
   componentDidUpdate(prevProps: Props) {
     if (prevProps.location.pathname !== this.props.location.pathname) {
       this.checkStateRoute();
     }
+    this.checkIfShouldMountWidgetViewerModal();
   }
 
   componentWillUnmount() {
     window.removeEventListener('beforeunload', this.onUnload);
   }
 
+  checkIfShouldMountWidgetViewerModal() {
+    const {
+      params: {widgetId},
+      organization,
+      dashboard,
+      location,
+    } = this.props;
+    if (defined(widgetId)) {
+      const widget = dashboard.widgets.filter(({id}) => id === `${widgetId}`)[0];
```

Comment:
If there's no widget, we should probably redirect/replace the url to the dashboard url + add a test for it

Reply:
I'm not very familiar with this part of our code... but in case there's no widget, shouldn't we inform the user? 'Widget Not Found' or something?

Reply:
I've updated to redirect the url to the dashboard url and raise the `Widget not found` error as a toast in this scenario!